src/components/AddPlaidBankAccount.js

```diff
@@ -0,0 +1,198 @@
+    plaidBankAccounts: {
+        loading: false,
+    },
+    onExitPlaid: () => {},
+    onSubmit: () => {},
+};
+
+class AddPlaidBankAccount extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.selectAccount = this.selectAccount.bind(this);
+
+        this.state = {
+            selectedIndex: undefined,
+            password: '',
+            isCreatingAccount: false,
+            institution: {},
+        };
+    }
+
+    componentDidMount() {
+        clearPlaidBankAccountsAndToken();
+        fetchPlaidLinkToken();
+    }
+
+    /**
+     * Get list of bank accounts
+     *
+     * @returns {Object[]}
+     */
+    getAccounts() {
+        return lodashGet(this.props.plaidBankAccounts, 'accounts', []);
+    }
+
+    selectAccount() {
+        const account = this.getAccounts()[this.state.selectedIndex];
+        this.props.onSubmit({
+            account, password: this.state.password, plaidLinkToken: this.props.plaidLinkToken,
+        });
+        this.setState({isCreatingAccount: true});
+    }
+
+    render() {
+        const accounts = this.getAccounts();
+        const options = _.chain(accounts)
+            .filter(account => !account.alreadyExists)
+            .map((account, index) => ({
+                value: index, label: `${account.addressName} ${account.accountNumber}`,
+            }))
+            .value();
+
+        return (
+            <>
+                {(!this.props.plaidLinkToken || this.props.plaidBankAccounts.loading)
+                    && (
+                        <View style={[styles.flex1, styles.alignItemsCenter, styles.justifyContentCenter]}>
+                            <ActivityIndicator size="large" />
+                        </View>
+                    )}
+                {!_.isEmpty(this.props.plaidLinkToken) && (
+                    <PlaidLink
+                        token={this.props.plaidLinkToken}
+                        onSuccess={({publicToken, metadata}) => {
+                            getPlaidBankAccounts(publicToken, metadata.institution.name);
+                            this.setState({institution: metadata.institution});
+                        }}
+                        onError={(error) => {
+                            console.debug(`Plaid Error: ${error.message}`);
+                        }}
+                        onExit={() => {
+                            // User prematurely exited the Plaid flow
+                            this.props.onExitPlaid();
+                        }}
+                    />
+                )}
+                {accounts.length > 0 && (
+                    <>
+                        <View style={[styles.m5, styles.flex1]}>
+                            {!_.isEmpty(this.props.text) && (
+                                <Text style={[styles.mb5]}>{this.props.text}</Text>
+                            )}
+                            {/* @TODO there are a bunch of logos to incorporate here to replace this name
```

Comment:
Created an issue here to follow up on this eventually https://github.com/Expensify/Expensify/issues/166782